core/unix/injector.c

```diff
@@ -259,16 +272,11 @@ pre_execve_ld_preload(const char *dr_path)
 #else
     setenv("LD_LIBRARY_PATH", ld_lib_path, true /*overwrite*/);
 
-    if (cur_preload != NULL) {
-        snprintf(ld_preload_libs, BUFFER_SIZE_ELEMENTS(ld_preload_libs), "%s %s",
-                 cur_preload, dr_ld_preload_libs);
-        NULL_TERMINATE_BUFFER(ld_preload_libs);
-        ld_preload = ld_preload_libs;
-    } else {
-        ld_preload = dr_ld_preload_libs;
-    }
+    snprintf(ld_lib_path, BUFFER_SIZE_ELEMENTS(ld_lib_path), "%s%s", preload_prefix,
```

Comment:
I think you've omitted the %c preload_delimiter.

A test would presumably catch this.  Though with -late not used much it seems ok to not put in the effort for it: would have to make some other lib to be preloaded and check it got in there too with a -late injection.

Reply:
Ah my bad: it's in the preload_prefix which is shared.